jax/api.py

```diff
@@ -112,6 +118,14 @@ def jit(fun: Callable, static_argnums: Union[int, Iterable[int]] = (),
       XLA's DeviceAssignment logic and is usually to use ``jax.devices()[0]``.
     backend: This is an experimental feature and the API is likely to change.
       Optional, a string representing the xla backend. 'cpu','gpu', or 'tpu'.
+    donate_argnums: Specify which arguments are "donated" to the computation.
```

Comment:
Should we document that this works only on TPUs? What happens on other platforms. I assume there is no error, just no aliasing.

Reply:
There is an error.

Reply:
I personally preferred this without the error, as it lets me run the same code for both local GPU & TPU cases. I now need to check devices & grab their platform to decide my pmap args!
(and I already expect that memory allocation behavior is different between GPU & TPU, so am not surprised that I can't run to the same efficiency on GPU.)

Reply:
+1, I'm not a real user so listen to Trevor more than me, but I like the idea of not needing to modify code to switch platforms (modulo low-level details like memory management...). Maybe issue a one-time warning if used on GPU?

Reply:
Per the discussion in the design review I've made consumption of donated buffers optional and as of this PR all we'll do is setup "must aliases" on TPU iff we find a match between the input and the output buffers.